web/packages/teleport/src/Assist/ConversationList/ConversationList.tsx

```diff
@@ -41,19 +41,45 @@ const Container = styled.div.attrs({ 'data-scrollbar': 'default' })`
 
 export function ConversationList(props: ConversationListProps) {
   const ref = useRef<HTMLDivElement>();
+  const scrollRef = useRef<HTMLDivElement>();
 
+  const shouldScroll = useRef(true);
   const scrolling = useRef<boolean>(false);
 
   const { conversations, selectedConversationMessages } = useAssist();
 
   function scroll() {
+    if (!shouldScroll.current) {
+      return;
+    }
+
     scrolling.current = true;
 
     ref.current.scrollIntoView({ behavior: 'smooth' });
 
     window.setTimeout(() => (scrolling.current = false), 1000);
   }
 
+  useEffect(() => {
+    if (!scrollRef.current) {
+      return;
+    }
+
+    function onscroll() {
+      const scrollPosition = scrollRef.current.scrollTop;
+      const maxScrollPosition =
+        scrollRef.current.scrollHeight - scrollRef.current.clientHeight;
+
+      // if the user has scrolled more than 50px from the bottom of the chat, assume they don't want the message list
+      // to auto scroll.
+      shouldScroll.current = scrollPosition > maxScrollPosition - 50;
+    }
+
+    scrollRef.current.addEventListener('wheel', onscroll);
+
+    return () => scrollRef.current.removeEventListener('wheel', onscroll);
+  }, [scrollRef.current]);
```

Comment:
Values held by refs are not valid deps for `useEffect`, because mutating them doesn't rerender the component, so the current dep array works only by accident. react-hooks/exhaustive-deps warns you about this. ;)

Since `scrollRef` is rendered only if `conversations.selectedId` is true, you could guard the effect with that check instead of `if (!scrollRef.current)`. Ideally, I feel like it'd be best to make the decision about rendering `<LandingPage />` vs the actual body of `ConversationList` somewhere sooner, so you don't have to worry about the fact that `scrollRef` is rendered conditionally.

See the pitfall in this section: https://react.dev/reference/react/useRef#referencing-a-value-with-a-ref

Reply:
Nice one, I moved the landing page to be earlier and rewrote the logic for scrolling to not use any refs. It responds even better now 😄